### PR TITLE
[Merged by Bors] - Introduce channels into CLI

### LINKED ISSUE #1663: CLI - Fluvio release channels
Currently in order to deploy version, we need to type:
```
 fluvio cluster start --image-version latest --help
```

This is error prone.  We need to have user to test development version quickly

### PR DESCRIPTION
**Pull request by tjtelan**
This adds the concept of separate `channels` in the CLI, with the ability to change between them.

3 default channels:
* `stable`
* `latest`
* `dev`

By default, we use the `stable` channel, but you can switch to another channel with:

`fluvio version switch <channel>`

To add a version channel, you need to create it first

For example, to create a channel for `0.9.15` run
`fluvio version create 0.9.15`

Then you can switch over to it with

`fluvio version switch 0.9.15`

---

Simplifies starting a cluster to `fluvio cluster start` for all channels (only if using the fluvio-channel binary)

Changes `fluvio-cli` to respect 3 environment variables to control cluster and plugin interactions
* `FLUVIO_RELEASE_CHANNEL`
* `FLUVIO_EXTENSIONS_DIR`
* `FLUVIO_IMAGE_TAG_STRATEGY`

Manual tests:
- Copy `fluvio-channel` binary `~/.fluvio/bin/fluvio` 
- Switch to stable channel `fluvio version switch stable`, start a cluster with `fluvio cluster start`
- Switch to latest, `fluvio version switch latest`, start a cluster with `fluvio cluster start`
- Verify the image that is used for pods is correct

Additionally, modifies `install.sh` to support the installation of `fluvio-channel` to `~/.fluvio/bin/fluvio`, and a target fluvio binary as `~/.fluvio/bin/fluvio-<channel>`

Resolves #1663

### CODE REVIEW

**Comment by tjtelan** [comment]
Looks like CLI works on `dev` channel. Need to do some testing to make sure `stable` and `latest` channels work as expected

**Review by sehz — CHANGES_REQUESTED** [review]
First pass:

Just several minor nits.

File: crates/fluvio-cli/src/derivedstream/describe.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@ use structopt::StructOpt;
 use crate::Result;
 
 /// Print details about a given DerivedStream
-#[derive(Debug, StructOpt)]
+#[derive(Debug, StructOpt, Clone)]
```
Comment (sehz) [inline crates/fluvio-cli/src/derivedstream/describe.rs:5]: why this need to be cloned?

**Review by sehz — CHANGES_REQUESTED** [review]
The major question is why this can't be in a separate crate?

**Comment by tjtelan** [comment]
Based on feedback, I'll be moving the `fluvio-channel` binary into its own crate.

Also I'm going to experiment with minimizing the channel selection logic in `fluvio` cli using environment variables that `fluvio-channel` will set. (Definitely leverage structopt to add this support into `fluvio`)

**Review by sehz — APPROVED** [review]
Looks great.  Nice Job!

**Comment by sehz** [comment]
rebase

**Comment by sehz** [comment]
zig issue is fixed.  can you rebase?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -156,6 +160,28 @@ jobs:
...
+            binary: fluvio-channel
+          - os: ubuntu-latest
+            rust: stable
+            rust-target: armv7-unknown-linux-gnueabihf
+            binary: fluvio-channel
+          - os: ubuntu-latest
+            rust: stable
+            rust-target: x86_64-pc-windows-gnu
+            binary: fluvio-channel.exe
+          - os: macos-latest
```
Comment (sehz) [inline .github/workflows/ci.yml:176]: this need to use `macos-11` with zig related changes

**Review by sehz — CHANGES_REQUESTED** [review]

**Comment by sehz** [comment]
windows build fails in staging:
```
   --> crates/fluvio-dataplane-protocol/src/record.rs:690:24
776
    |
777
690 |     use fluvio_future::file_slice::AsyncFileSlice;
778
    |                        ^^^^^^^^^^ could not find `file_slice` in `fluvio_future`
```

**Pull request closed without merging**